instrumentation/apache-camel-2.20/src/main/java/io/opentelemetry/instrumentation/auto/apachecamel/decorators/HttpSpanDecorator.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.auto.apachecamel.decorators;
+/*
+ * Includes work from:
+ * Copyright Apache Camel Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import io.opentelemetry.trace.Span;
+import io.opentelemetry.trace.attributes.SemanticAttributes;
+import org.apache.camel.Endpoint;
+import org.apache.camel.Exchange;
+
+class HttpSpanDecorator extends BaseSpanDecorator {
+
+  private static final String POST_METHOD = "POST";
+  private static final String GET_METHOD = "GET";
+
+  public static String getHttpMethod(Exchange exchange, Endpoint endpoint) {
+    // 1. Use method provided in header.
+    Object method = exchange.getIn().getHeader(Exchange.HTTP_METHOD);
+    if (method instanceof String) {
+      return (String) method;
+    }
+
+    // 2. GET if query string is provided in header.
+    if (exchange.getIn().getHeader(Exchange.HTTP_QUERY) != null) {
+      return GET_METHOD;
+    }
+
+    // 3. GET if endpoint is configured with a query string.
+    if (endpoint.getEndpointUri().indexOf('?') != -1) {
+      return GET_METHOD;
+    }
+
+    // 4. POST if there is data to send (body is not null).
+    if (exchange.getIn().getBody() != null) {
+      return POST_METHOD;
+    }
+
+    // 5. GET otherwise.
+    return GET_METHOD;
+  }
+
+  @Override
+  public String getOperationName(Exchange exchange, Endpoint endpoint) {
+    // Based on HTTP component documentation:
+    return getHttpMethod(exchange, endpoint);
```

Comment:
Is it possible to use the route for operation name? (ass suggested in [semantic conventions](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/semantic_conventions/http.md#name))

Reply:
Theoretically - yes, obviously. Practically- not that much.
1. Http decorator serves multiple components - ahc, http, jetty, undertow.
2. Component route can include all GET parameters (query) - would yield high cardinality
3. Same decorator is used for both sides (SERVER, CLIENT) - so being conservative (always use HTTP method) won

At the same time if there is a consensus that using method in all cases is wrong, I'm open to discuss alternatives, like:
- adding more http decorators to handle different cases separately
- removing query
This however won't allow us to eg sanitize path fragments that are values and inevitably yields high cardinality.